src/ValueObject/Application/File.php

```diff
@@ -143,6 +143,14 @@ public function getOldTokens(): array
         return $this->oldTokens;
     }
 
+    /**
+     * @param mixed[] $oldTokens
+     */
+    public function setOldTokens(array $oldTokens): void
+    {
+        $this->oldTokens = $oldTokens;
+    }
```

Comment:
We need to avoid using tokens as much as we can.
The printer could be a way to handle the responsibility for printing. We already modify spaces around double colons there.

Reply:
I will try.

Reply:
@TomasVotruba I tried changing to use `Lexer` instead of `Emulative\Lexer` on https://github.com/TomasVotruba/group-use-test and it seems the result is same, it possibly the issue on `RichParser` itself?

Reply:
@samsonasik That might be the case, but whas it set-up different there? In the end it should be using php-parser

Reply:
not sure how to configure it in the parser.neon, we currently use RichParser with `@currentPhpVersionLexer` pointed to `Lexer\Emulative`:

https://github.com/rectorphp/rector-src/blob/3e66b4a6348e2d106dc714760c946003bdc9807d/config/phpstan/parser.neon#L23-L28